src/components/Proposal/Create/PluginForms/CreateUnknownGenericPluginProposal.tsx

```diff
@@ -72,12 +89,23 @@ class CreateGenericPlugin extends React.Component<IProps, IStateProps> {
       () => Object.assign(this.currentFormValues, this.state),
       (formValues: IFormValues, firstTime: boolean) => {
         this.currentFormValues = formValues;
-        if (firstTime) { this.state = { tags: formValues.tags }; }
-        else { this.setState({ tags: formValues.tags }); }
+        if (firstTime) {
+          Object.assign(this.state, {
+            tags: formValues.tags, abiInputs: this.state.abiInputs, callData: this.state.callData,
+          });
+        }
+        else { this.setState({ tags: formValues.tags, abiInputs: [], callData: "" }); }
       },
       this.props.showNotification);
   }
 
+  async componentDidMount() {
+    const contractToCall = (this.props.pluginState as IGenericPluginState).pluginParams.contractToCall;
+    const ABI_DATA = await getABIByContract(contractToCall);
+    const ABI_METHODS = extractABIMethods(ABI_DATA);
```

Comment:
how about combine `getABIByContract` and `extractABIMethods` to one function `extractABIMethodsByContract` ?

Reply:
Like this we keep the functions separate and if we ever need to get an ABI or to extract methods from an existing ABI there will be no need to change any code.